internal/kafka/test/integration/admin_kafka_test.go

```diff
@@ -1312,19 +1313,24 @@ func TestAdminKafka_Update(t *testing.T) {
 	defer tearDown()
 	db := test.TestServices.DBFactory.New()
 	// create a dummy cluster and assign a kafka to it
-	cluster := &api.Cluster{
-		Meta: api.Meta{
+	cluster := mockclusters.BuildCluster(func(cluster *api.Cluster) {
+		cluster.Meta = api.Meta{
 			ID: api.NewID(),
-		},
-		ClusterID:          api.NewID(),
-		MultiAZ:            true,
-		Region:             "baremetal",
-		CloudProvider:      "baremetal",
-		Status:             api.ClusterReady,
-		IdentityProviderID: "some-id",
-		ClusterDNS:         "some-cluster-dns",
-		ProviderType:       api.ClusterProviderStandalone,
-	}
+		}
+		cluster.ProviderType = api.ClusterProviderStandalone
+		cluster.SupportedInstanceType = api.AllInstanceTypeSupport.String()
+		cluster.ClientID = "some-client-id"
+		cluster.ClientSecret = "some-client-secret"
+		cluster.ClusterID = api.NewID()
+		cluster.Region = "baremetal"
+		cluster.CloudProvider = "baremetal"
+		cluster.MultiAZ = true
+		cluster.Status = api.ClusterReady
+		cluster.ProviderSpec = api.JSON{}
+		cluster.ClusterSpec = api.JSON{}
+		cluster.ClusterDNS = "some-cluster-dns"
+		cluster.IdentityProviderID = "some-identity-provider-id"
+	})
```

Comment:
This code is repeated a few times. We could enhance the `BuildCluster` to use the `functional options pattern` and provide a `WithDefailtTestValues` so that this would become: 
```
cluster := mockclusters.BuildCluster(
   WithDefaultTestValues(),
   WithMyCustomValue1(),  // fictional method 
   WithMyCustomValue2(), // fictional method
)
```

Reply:
Thanks for the review. 

I am happy for that to be done outside the scope of this PR.